### PR TITLE
Update hub.py

### LINKED ISSUE #4862: KeyError: 'version' when installing a plugin
## 🐛 Bug

KeyError: 'version' when installing a plugin (that is on pypi but not yet on napari hub).
This error is maybe the reason why the warning message  'When installing/uninstalling npe2 plugins, you must restart napari for UI changes to take effect.' is not displaying...

## To Reproduce

1. Install the `napari-deepfinder` plugin from the UI by typing `napari-deepfinder` in the text bar.
2. The 

```
KeyError                                  Traceback (most recent call last)
File /home/cnstt/test-nap/venv1/lib/python3.9/site-packages/superqt/utils/_qthreading.py:616, in create_worker.<locals>.reraise(e=KeyError('version'))
    615 def reraise(e):
--> 616     raise e
        e = KeyError('version')

File /home/cnstt/test-nap/venv1/lib/python3.9/site-packages/superqt/utils/_qthreading.py:176, in WorkerBase.run(self=<napari._qt.qthreading.GeneratorWorker object>)
    174     warnings.filterwarnings("always")
    175     warnings.showwarning = lambda *w: self.warned.emit(w)
--> 176     result = self.work()
        self = <napari._qt.qthreading.GeneratorWorker object at 0x7faf71497790>
    177 if isinstance(result, Exception):
    178     if isinstance(result, RuntimeError):
    179         # The Worker object has likely been deleted.
    180         # A deleted wrapped C/C++ object may result in a runtime
    181         # error that will cause segfault if we try to do much other
    182         # than simply notify the user.

File /home/cnstt/test-nap/venv1/lib/python3.9/site-packages/superqt/utils/_qthreading.py:443, in GeneratorWorker.work(self=<napari._qt.qthreading.GeneratorWorker object>)
    441 try:
    442     input = self._next_value()
--> 443     output = self._gen.send(input)
        self = <napari._qt.qthreading.GeneratorWorker object at 0x7faf71497790>
        input = None
        output = (PackageMetadata(metadata_version='1.0', name='zarpaint', version='0.2.0', dynamic=None, platform=None, supported_platform=None, summary='Paint segmentations directly to on-disk/remote zarr arrays', description=None, description_content_type=None, keywords=None, home_page='https://github.com/jni/zarpaint', download_url=None, author='Abigail McGovern and Juan Nunez-Iglesias', author_email=None, maintainer=None, maintainer_email=None, license='BSD-3-Clause', classifier=None, requires_dist=None, requires_python=None, requires_external=None, project_url=None, provides_extra=None, provides_dist=None, obsoletes_dist=None), True)
        self._gen = <generator object iter_hub_plugin_info at 0x7faf714d2c80>
    444     self.yielded.emit(output)
    445 except StopIteration as exc:

File /home/cnstt/test-nap/venv1/lib/python3.9/site-packages/napari/plugins/hub.py:101, in iter_hub_plugin_info(skip={}, conda_forge=False)
     94 futures = [
     95     executor.submit(hub_plugin_info, name, conda_forge=conda_forge)
     96     for name in sorted(plugins)
     97     if name not in skip
     98 ]
    100 for future in as_completed(futures):
--> 101     info, is_available_in_conda_forge = future.result()
        future = <Future at 0x7faf7270c9d0 state=finished raised KeyError>
        info, is_available_in_conda_forge = (PackageMetadata(metadata_version='1.0', name='zarpaint', version='0.2.0', dynamic=None, platform=None, supported_platform=None, summary='Paint segmentations directly to on-disk/remote zarr arrays', description=None, description_content_type=None, keywords=None, home_page='https://github.com/jni/zarpaint', download_url=None, author='Abigail McGovern and Juan Nunez-Iglesias', author_email=None, maintainer=None, maintainer_email=None, license='BSD-3-Clause', classifier=None, requires_dist=None, requires_python=None, requires_external=None, project_url=None, provides_extra=None, provides_dist=None, obsoletes_dist=None), True)
        info = PackageMetadata(metadata_version='1.0', name='zarpaint', version='0.2.0', dynamic=None, platform=None, supported_platform=None, summary='Paint segmentations directly to on-disk/remote zarr arrays', description=None, description_content_type=None, keywords=None, home_page='https://github.com/jni/zarpaint', download_url=None, author='Abigail McGovern and Juan Nunez-Iglesias', author_email=None, maintainer=None, maintainer_email=None, license='BSD-3-Clause', classifier=None, requires_dist=None, requires_python=None, requires_external=None, project_url=None, provides_extra=None, provides_dist=None, obsoletes_dist=None)
        is_available_in_conda_forge = True
    102     if info and info not in already_yielded:
    103         already_yielded.append(info)

File /usr/lib64/python3.9/concurrent/futures/_base.py:439, in Future.result(self=None, timeout=None)
    437     raise CancelledError()
    438 elif self._state == FINISHED:
--> 439     return self.__get_result()
        self = None
    441 self._condition.wait(timeout)
    443 if self._state in [CANCELLED, CANCELLED_AND_NOTIFIED]:

File /usr/lib64/python3.9/concurrent/futures/_base.py:391, in Future.__get_result(self=None)
    389 if self._exception:
    390     try:
--> 391         raise self._exception
        self = None
    392     finally:
    393         # Break a reference cycle with the exception in self._exception
    394         self = None

File /usr/lib64/python3.9/concurrent/futures/thread.py:58, in _WorkItem.run(self=None)
     55     return
     57 try:
---> 58     result = self.fn(*self.args, **self.kwargs)
        self = None
     59 except BaseException as exc:
     60     self.future.set_exception(exc)

File /home/cnstt/test-nap/venv1/lib/python3.9/site-packages/napari/plugins/hub.py:47, in hub_plugin_info(name='napari-deepfinder', min_dev_status=3, conda_forge=False)
     44 except error.HTTPError:
     45     return None, False
---> 47 version = info["version"]
        info = {'conda': [], 'display_name': '', 'plugin_types': [], 'reader_file_extensions': [], 'writer_file_extensions': [], 'writer_save_layers': []}
     48 norm_name = normalized_name(info["name"])
     49 is_available_in_conda_forge = True

KeyError: 'version'


```

## Expected behavior

This error should not happen!

## Environment

 napari: 0.4.16
Platform: Linux-5.18.10-100.fc35.x86_64-x86_64-with-glibc2.34
System: Fedora Linux 35 (Workstation Edition)
Python: 3.9.13 (main, Jun 9 2022, 00:00:00) [GCC 11.3.1 20220421 (Red Hat 11.3.1-2)]
Qt: 5.15.2
PyQt5: 5.15.7
NumPy: 1.23.1
SciPy: 1.8.1
Dask: 2022.7.1
VisPy: 0.10.0

OpenGL:
- GL version: 4.6.0 NVIDIA 515.48.07
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1920x1200, scale 1.0

Plugins:
- console: 0.0.4
- napari-svg: 0.1.6
- scikit-image: 0.4.16

### PR DESCRIPTION
**Pull request by cnstt**
# Description
Fixes #4862 

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References
closes #4862

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

**Comment by cnstt** [comment]
ping @Czaki @tlambert03 This is maybe not the most clean code (if you have a way to check if all elements of the dict are "empty" I would happily replace my code), but it fixes a bug ;)

File: napari/plugins/hub.py [hunk 1]
```diff
@@ -44,6 +44,18 @@ def hub_plugin_info(
...
+    # simply return None, False like the above except
+    if info == {
+        'conda': [],
+        'display_name': '',
+        'plugin_types': [],
+        'reader_file_extensions': [],
+        'writer_file_extensions': [],
+        'writer_save_layers': [],
+    }:
+        return None, False
```
Comment (Czaki) [inline napari/plugins/hub.py]: I do not like this. The conda response may change in the future, and this check will be wrong. I would prefer to return these values if, for example `"version"` is missed from dict.

Comment (cnstt) [reply napari/plugins/hub.py]: I had the same feeling when I wrote these lines of code... Will need to rewrite this then ;)

File: napari/plugins/hub.py [hunk 2]
```diff
@@ -44,6 +44,18 @@ def hub_plugin_info(
...
+    # If the napari hub returns an empty info dict,
+    # simply return None, False like the above except
+    if info == {
+        'conda': [],
+        'display_name': '',
+        'plugin_types': [],
+        'reader_file_extensions': [],
+        'writer_file_extensions': [],
+        'writer_save_layers': [],
+    }:
```
Comment (neuromusic) [inline napari/plugins/hub.py]:
```suggestion
    if 'name' not in info:
```

File: napari/plugins/hub.py (hunk 1 above)
Comment (neuromusic) [reply napari/plugins/hub.py]: added a suggestion

File: napari/plugins/hub.py (hunk 2 above)
Comment (tlambert03) [reply napari/plugins/hub.py]: just to be safe, should it maybe be everything that gets directly indexed?
`'name', 'version', 'authors', 'summary', 'license'`?  or is there no chance that if name makes it in, the other ones don't make it in?

**Comment by cnstt** [comment]
I rewrote my bugfix thanks to you suggestions @neuromusic and @tlambert03, thanks!

File: napari/plugins/hub.py [hunk 3]
```diff
@@ -46,7 +46,7 @@ def hub_plugin_info(
 
     # If the napari hub returns an info dict missing the required keys,
     # simply return None, False like the above except
-    if not {'name', 'version', 'authors', 'summary', 'license'} <= info.keys():
+    if not {'name', 'version', 'authors', 'summary', 'license', 'project_site'} <= info.keys():
```
Comment (tlambert03) [inline napari/plugins/hub.py]: nice :) 👍

**Review by tlambert03 — APPROVED** [review]

**Review by Czaki — APPROVED** [review]

**Pull request merged**